src/radical/pilot/staging_directives.py

```diff
@@ -145,7 +145,7 @@ def complete_url(path   : str,
     certain locations, namely relative to
 
         * `client://`  : the client's working directory
-        * `endpoint://`: the root file system on the target resource
+        * `endpoint://`: the root of the target resource's file system
```

Comment:
A previous comment qualified this definition with "if possible". Can you clarify what happens in the not-possible scenario?

Reply:
We access the system via some protocol - the service providing the protocol could do whatever, and we have no way of asserting that the accessed file system location is indeed the file system root.  It could be a sandbox the service lives in, for example.

Reply:
Would it be fair to say something like the "root-most directory reachable from the filesystem endpoint defined for the resource"?

Reply:
I don't think that would add any clarity, really, and trying to detail this more depending on the respective protocol is of little use.  I expect that a user or application who gets an URL with a certain protocol knows how to use that protocol.

Reply:
But there is no indication of the protocol here. In order to know what `root` should mean to the application programmer, we need to know what it means to RP. It is not necessarily obvious whether this scheme is derived from the `filesystem_endpoint` field or from `cd /` in the pilot/task environment.

That may be a lot to document right here. Elsewhere, I have suggested consolidating the staging directives documentation to a module docstring and providing more explanation. But if `endpoint:///` is strongly defined to be equivalent to the result of `radical.pilot.utils.misc.get_resource_fs_url()`, a "see also" would probably be sufficient.

Reply:
> But if endpoint:/// is strongly defined...

it is.

Reply:
> > But if endpoint:/// is strongly defined...
> 
> it is.

This contradicts https://github.com/radical-cybertools/radical.pilot/issues/2700#issuecomment-1262907802

If `radical.pilot.utils.misc.get_resource_fs_url()` returns the `filesystem_endpoint`, and `filesystem_endpoint` does not lead to the root of a filesystem, then `endpoint:///` cannot both be an absolute path and equal to the result of `radical.pilot.utils.misc.get_resource_fs_url()`.

What have I misunderstood?

Reply:
Well, yes, apart from any service containerization we cannot do nothing about - but again, that is a bit academic.  

I think we are running a bit in a circle here.

So, `fs_endpoint + cd / == endpoint:///`.  This usually points to the root of the systems' file system - and where it does not, we can not do anything about it.

Reply:
That is because an URL does not have a notion of a relative path.  The need for `cd /` is an artifact of the service which interprets that URL and is outside the scope of what we can define.  We happen to handle it that way because SAGA works that way - but other tools may (and do) work differently.  The exposed URL is, per your use case, expected to be used outside of RP, and I just don't want to make assumptions on how it is used.

For example, using that URL with `curl` or `rsync` does not allow a notion of `cd /`, and there is nothing we can do about it...